library/std/src/panic.rs

```diff
@@ -503,21 +502,18 @@ pub fn get_backtrace_style() -> Option<BacktraceStyle> {
         return Some(style);
     }
 
-    let format = crate::env::var_os("RUST_BACKTRACE")
-        .map(|x| {
-            if &x == "0" {
-                BacktraceStyle::Off
-            } else if &x == "full" {
+    let format = match crate::env::var_os("RUST_BACKTRACE") {
+        Some(x) if &x == "0" => BacktraceStyle::Off,
+        Some(x) if &x == "full" => BacktraceStyle::Full,
+        Some(_) => BacktraceStyle::Short,
+        None => {
+            if crate::sys::FULL_BACKTRACE_DEFAULT {
                 BacktraceStyle::Full
             } else {
                 BacktraceStyle::Short
             }
```

Comment:
One more cleanup to be had:

```suggestion
        None if crate::sys::FULL_BACKTRACE_DEFAULT => BacktraceStyle::Full,
        None => BacktraceStyle::Short,
```

(don't apply directly because GH doesn't seem to let me highlight the closing brace)

With the above this LGTM, please squash into one commit and I'll approve once CI is green.

Reply:
Thanks a lot.

Reply:
The last case should actually be `BacktraceStyle::Off`, not `Short`

Reply:
sorry my lack of attention...

Reply:
No worries, I missed it too at first :)